src/except.cpp

```diff
@@ -108,16 +127,46 @@ std::string to_string(const Unit &unit) {
 }
 
 std::string to_string(const Variable &variable) {
-  std::string s("Variable(");
+  std::string prefix =
+      variable.isCoord() ? "Coord" : variable.isData() ? "Data" : "Attr";
+  std::string s = prefix + "Variable(";
   s += to_string(variable.tag()) + ", " + variable.name() + ")";
   return s;
 }
 
+std::string to_string_variable(const Dataset &dataset,
+                               const std::set<size_t> &indexes) {
+  std::string out;
+  for (auto idx : indexes) {
+    out += "\n" + to_string(dataset[idx]);
+  }
+  return out;
+}
+
 std::string to_string(const Dataset &dataset) {
   std::string s("Dataset with ");
   s += std::to_string(dataset.size()) + " variables";
+  s += "\nDimensions : " + to_string(dataset.dimensions());
+  // The following is peformed to allow variables to be sorted into catagories
+  // of coordinate, data and attribute as part of output
+  auto coordIndexes =
+      filter_variables(dataset, [](const ConstVariableSlice &var) -> auto {
+        return var.isCoord();
+      });
+  s += to_string_variable(dataset, coordIndexes);
```

Comment:
Why not just

```cpp
for (const auto &var: dataset)
  if (var.isCoord())
    s += '\n' + to_string(var);
```
?

It saves 2 lines here (times 3), plus 20 lines saved for two helper methods that are not necessary.

Reply:
xarray representations group the variables into types on output.

Reply:
This would not allow us to group the categories of variable type. AFAIK, the small vector of variables does not keep them in any particular sorted order. I'm aware of how xarray outputs it's representations. If you want to yield them in an un-ordered way and use the prefix `Coord.` `Data.` etc, then what you suggest would work fine, otherwise you need a way to get all tags of the same type out at once.

Reply:
I don't think we have to worry about sorting at this point. Probably things should just be sorted by name (within the coord/data/attr category), since the tag is actually the secondary part of the identifier.

Reply:
Ok, in that case I'll report the tag and name but yield them in the same order they are stored. That will simplify the code a lot.

Reply:
We should probably consider to refactor the interface at some point: Having the name before the tag actually makes more sense.